typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+import { ServiceID, NetworkContext, Network, ServiceContext, PortBinding, ContainerRunConfig, StaticFileID, ContainerCreationConfig, ContainerCreationConfigBuilder, ContainerRunConfigBuilder } from "kurtosis-core-api-lib";
+import { TestConfigurationBuilder } from "kurtosis-testsuite-api-lib"; //TODO (Ali)
+import { Result, err, ok, ResultAsync, errAsync, okAsync } from "neverthrow";
+import * as log from "loglevel";
+import { DatastoreClient } from "../../../datastore/datastore_service_client/datastore_client";
+import * as fs from 'fs';
+import { APIClient, Person } from "../../../api/api_service_client/api_client"; //TODO - Extract api client to example-microservices 
+
+
+const DATASTORE_IMAGE: string = "kurtosistech/example-microservices_datastore";
+const DATASTORE_SERVICE_ID: ServiceID = "datastore";
+const DATASTORE_PORT: number = 1323;
+
+const API_SERVICE_IMAGE: string = "kurtosistech/example-microservices_api";
+const API_SERVICE_ID: ServiceID = "api";
+const API_SERVICE_PORT: number = 2434;
+
+const WAIT_FOR_STARTUP_DELAY_MILLISECONDS: number = 1000;
+const WAIT_FOR_STARTUP_MAX_POLLS: number = 15;
+
+const TEST_PERSON_ID: number = 23;
+const TEST_NUM_BOOKS_READ: number = 3;
+
+const CONFIG_FILE_KEY: string  = "config-file";
+
+
+class DatastoreConfig {
+	private readonly datastoreIp: string;
+	private readonly datastorePort: number;
+	
+	constructor(datastoreIp: string, datastorePort: number) {
+		this.datastoreIp = datastoreIp;
+		this.datastorePort = datastorePort;
+	}
+}
+
+export class BasicDatastoreAndApiTest {
+    private readonly datastoreImage: string;
+    private readonly apiImage: string;
+    
+    constructor(datastoreImage: string, apiImage: string) {
+		this.datastoreImage = datastoreImage;
+		this.apiImage = apiImage;
+    }
+	
+	public configure(builder: TestConfigurationBuilder): void {
+		builder.withSetupTimeoutSeconds(60).withRunTimeoutSeconds(60); //TODO (Ali) - allowed since typescript gives direct reference
+    }
+
+	public async setup(networkCtx: NetworkContext): Promise<Result<Network, Error>> { //TODO (Ali) - async?
+		
+		const [datastoreContainerCreationConfig, datastoreRunConfigFunc]: [ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>] = await getDatastoreServiceConfigurations(); //TODO (Ali) - maybe Result here
```

Comment:
Yeah, the typings on this get real messy real quickly. `getDatastoreServiceConfigurations` is only two lines and only used once, so I've actually wanted to just inline its contents and get rid of it. You could do that here with both the Typescript and Go versions